kotlin-analysis-core/src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/AnalysisExecutor.kt

```diff
@@ -0,0 +1,48 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.ex.ProjectManagerEx
+import org.jetbrains.research.ml.kotlinAnalysis.util.getSubdirectories
+import java.nio.file.Path
+
+/**
+ * Abstract class for analysis executor which provides interface for execution analysis
+ * for each project in given dataset.
+ */
+abstract class AnalysisExecutor {
+
+    /** Executes the analysis of the given [project][Project]. */
+    abstract fun analyse(project: Project)
+
+    /**
+     * Finishes the task execution. Should contain close functions of all writers and readers
+     * and called after all project analysis execution finished.
+     */
+    abstract fun finish()
```

Comment:
I'm not sure, it might be discussed. Now it seems that every time we create an Executor we do the same actions in the `finish` function. Probably we can have some list\map or something like this with writers, init them in the constructor and implement the `finish` function here? We can create a meeting to discuss it. What do you think?

Reply:
Yes. I think, it would be great! Because it is one of key point of instrument usage workflow.